extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java

```diff
@@ -162,8 +163,18 @@ private String createContainerImage(ContainerImageConfig containerImageConfig, D
             OutputTargetBuildItem out, ImageIdReader reader, boolean forNative, boolean pushRequested,
             PackageConfig packageConfig) {
 
+        var useBuildx = dockerConfig.buildx.useBuildx();
+        var pushImages = pushRequested || containerImageConfig.isPushExplicitlyEnabled();
```

Comment:
What's with `pushImages`?

Reply:
I'm not sure exactly what you mean by what's with it, but basically pushing images is different based on if you're using buildx or not. If not using any of the buildx params, then the flow is as it was before:

1. Build the image (`docker build`)
2. Apply any tags (`docker tag|)
3. Push the image and all tags (`docker push`)

If using any of the buildx options, the tagging & pushing happens as part of the `docker buildx build` command via the added `-t` and `--push` params.

This is because when using buildx with more than one platform the resulting images are not loaded into the `docker images`, so therefore a `docker tag` or `docker push` will not work.

I'll re-read the documentation I wrote too and make sure everything is covered there.

Is that the explanation you were looking for?

Reply:
Thanks for the explanation. This should probably also be in comments in the source

Reply:
It kinda already is. There are comments scattered throughout the code at various code branches (where the conditions are checked).

Would you rather me have one long multi-line comment right at the beginning with this explanation?

Reply:
> Would you rather me have one long multi-line comment right at the beginning with this explanation?

Yeah, that would be great

Reply:
Done. Just pushed & rebased